llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -136,11 +136,32 @@ template <typename LTy, typename RTy> struct match_combine_or {
   }
 };
 
+template <typename LTy, typename RTy> struct match_combine_and {
+  LTy L;
+  RTy R;
+
+  match_combine_and(const LTy &Left, const RTy &Right) : L(Left), R(Right) {}
+
+  template <typename ITy> bool match(ITy *V) const {
+    if (L.match(V))
+      if (R.match(V))
+        return true;
+    return false;
```

Comment:
Can this just be 
```suggestion
    return L.match(V) && R.match(V);
```
?

Reply:
Done in f2cc512e7354acf80e51af363bdffe3df3d3e954